### PR TITLE
Test reverse relationships, small documentation update

### PR DESCRIPTION
**Pull request by bryanhelmig**
Works as expected! But the reverse cases would need some support added.

### CODE REVIEW

**Comment by bryanhelmig** [comment]
Looks like I'd have to do some similar contribute to class patching to handle some of the reverse, or even Generic relationships. Depends on how foolproof :tm: we'd need this library.

I'll take a looksie again tonight.

**Comment by charettes** [comment]
yeah for the reverse fk you should be able to use `related_accessor_class`

https://github.com/django/django/blob/c2d0f8c084456b5073252a91eeb09ab3d7453b18/django/db/models/fields/related.py#L459

Just like we did with `forward_related_accessor_class`

https://github.com/charettes/django-seal/blob/b695fcb1fe37f52a76be2841ed1531dc74b9cbc6/seal/models.py#L15

We'll also want to add support for `OneToOneField` by defining `ReverseOneToOneDescriptor` and `ForwardOneToOneDescriptor` subclasses in `sealable_accessor_classes`.

https://github.com/charettes/django-seal/blob/b695fcb1fe37f52a76be2841ed1531dc74b9cbc6/seal/related.py#L44-L47

**Comment by charettes** [comment]
alright, added reverse foreign keys and initial o2o support in #3. you might want to rebase on top of it. I incorporated your `test_sealed_instance_reverse_related_attribute_access` into it.

File: tests/test_models.py [hunk 1]
```diff
@@ -40,9 +40,23 @@ def test_sealed_instance_reverse_parent_link_access(self):
         with self.assertRaisesMessage(SealedObject, message):
             instance.sealion_ptr
 
+    def test_sealed_instance_reverse_related_attribute_access(self):
+        instance = Location.from_db('default', ['latitude', 'longitude'], [1.2, 3.4])
+        instance._state.sealed = True
+        message = "Cannot fetch many-to-many field visitors on a sealed object."
+        with self.assertRaisesMessage(SealedObject, message):
+            instance.visitors.all()
+
```
Comment (charettes) [inline tests/test_models.py]: This can be dropped, I included it in `test_sealed_instance_reverse_foreign_key_access` above.

**Comment by bryanhelmig** [comment]
Ah, I figured the contribute_to_related_class would be easy enough to knock out but looks it trying to touch `m2m.related_model._meta` upsets the app registry! cc @charettes

File: tests/test_models.py [hunk 2]
```diff
@@ -46,3 +46,10 @@ def test_sealed_instance_m2m_access(self):
         message = "Cannot fetch many-to-many field previous_locations on a sealed object."
         with self.assertRaisesMessage(SealedObject, message):
             instance.previous_locations.all()
+
+    def test_sealed_instance_reverse_m2m_access(self):
+        instance = Location.from_db('default', ['latitude', 'longitude'], [1.2, 3.4])
```
Comment (charettes) [inline tests/test_models.py]: It looks like the test was failing because deferred instances must always have a pk.

**Comment by charettes** [comment]
Superseded by #5.

**Pull request closed without merging**